CMakeLists.txt

```diff
@@ -59,6 +59,7 @@ cmake_dependent_option(
     USE_GLOO "Use Gloo" ON
     "BUILD_CAFFE2" OFF)
 option(USE_GLOO_IBVERBS "Use Gloo IB verbs for distributed support" OFF)  # New option
+option(USE_HIP "Use HIP" ON)
```

Comment:
Can you set "option(USE_HIP "Use HIP" ON)" to OFF? It's best to assume that users don't have HIP on their system unless it has been detected.

Reply:
@Jorghi12 Default to ON is ok, we just need to add some logic to check whether the hip headers and libraries are actually available in the system and change it to OFF if not.

Reply:
That's what I meant. You can use the logic for checking hip headers / libraries to set it to ON.